testy.go

```diff
@@ -0,0 +1,112 @@
+package testy
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+)
+
+type testy struct {
+	// TODO hierarchy? need an ordered map, this is shortly on TODO list and also why 1.18beta in use for generics
+	tests []*Test
+}
+
+var instance testy
+
+// TestingT is a subset of testing.T that we have to implement for non-`go test` runs.
+//
+// TODO flesh this out with more useful stuff from testing.T
+type TestingT interface {
+	Fail()
+	FailNow()
+	Fatal(args ...interface{})
+	Fatalf(format string, args ...interface{})
+	Errorf(format string, args ...interface{})
+	Helper()
+	Log(args ...interface{})
+	Logf(format string, args ...interface{})
+	Name() string
+}
+
+type Tester func(t TestingT)
+
+type Test struct {
```

Comment:
debating naming this `TestCase` so `RegisterTest` can be named `Test`. This might also not need to be exported.

Reply:
Would `Result` be fitting? That seems to be how it's used in the Run func.

Reply:
It's the result and the definition of the test to be run.

Reply:
I do like the name `RegisterTest` for that function as `Test` might imply that the test will be run immediately. The only other suggestion I have for this is maybe having test result and test definition structs and compose the test case struct with those. Names are hard.